quantum_systems/spatial_orbital_system.py

```diff
@@ -0,0 +1,147 @@
+from quantum_systems import QuantumSystem, GeneralOrbitalSystem
+
+
+class SpatialOrbitalSystem(QuantumSystem):
+    r"""Quantum system containing orbital matrix elements, i.e., we only keep
+    the spatial orbitals as they are degenerate in each spin direction. We have
+
+    .. math:: \psi(x, t) = \psi(\mathbf{r}, t) \sigma(m_s),
+
+    where :math:`x = (\mathbf{r}, m_s)` is a generalized coordinate of position
+    and spin, and :math:`sigma(m_s)` is either :math:`\alpha(m_s)` or
+    :math:`\beta(m_s)` as the two-dimensional spin basis states. This means
+    that we only store :math:`psi(mathbf{r}, t)`.
+
+    Parameters
+    ----------
+    n : int
+        Number of occupied basis functions. Note that this should correspond to
```

Comment:
A bit unclear, as number of particles `!=` number of occupied basis functions. I think it should be called number of particles, maybe by another name (`N`?) and then changed to `n = N // 2` internally. 

Alternatively one can send in the number of basis functions `n`, and then never talk about particles.

Reply:
Hmm, agreed. In practice `n` is interpreted as the number of particles upon initialization, but internally changed to `n // 2`. Perhaps just stating that `n` is the number of particles in the documentation is enough? We can also add a doctest demonstrating this, viz.
```python
>>> n = 4
>>> l = 20
>>> dim = 2
>>> spas = SpatialOrbitalSystem(n, RandomBasisSet(l, dim))
>>> spas.n == n // 2
True
```